app/models/contribution.rb

```diff
@@ -51,4 +59,10 @@ def update_reviewer(reviewer_id)
   scope :active_engineers, -> { joins(:user).merge(User.engineer.active) }
   scope :valid_pull_requests, -> { where.not(state: :refused) }
   scope :without_pr_state, ->(state) { where.not(pr_state: state) }
+
+  private
+
+  def rejected_reason_presence
+    errors.add(:rejected_reason, :must_be_blank, state: Contribution.human_attribute_name("state/#{state}")) if (approved? || received?) && !rejected_reason.blank?
```

Comment:
this line is too long WDYT about using regular if syntax instead the inline?

Reply:
I think it's a good idea. Changed